examples/py_triple.rs

```diff
@@ -0,0 +1,411 @@
+        &self,
+        mut layouter: impl Layouter<Fp>,
+        a: Fp,
+        b: Fp,
+    ) -> Result<(Alloc, Alloc), Error> {
+        let config = self.config();
+        let mut a_alloc: Option<Alloc> = None;
+        let mut b_alloc: Option<Alloc> = None;
+
+        layouter.assign_region(
+            || "load private inputs",
+            |mut region| {
+                let row_offset = 0;
+                let a_cell = region.assign_advice(
+                    || "private input 'a'",
+                    config.l_col,
+                    row_offset,
+                    || Ok(a),
+                )?;
+                let b_cell = region.assign_advice(
+                    || "private input 'b'",
+                    config.r_col,
+                    row_offset,
+                    || Ok(b),
+                )?;
+                // Note that no arithmetic is performed here, all we are doing is allocating the
+                // initial private wire values (i.e. private values which are not the output of any
+                // gate), thus there is no selector enabled in this row.
+                a_alloc = Some(Alloc { cell: a_cell, value: a });
+                b_alloc = Some(Alloc { cell: b_cell, value: b });
+                Ok(())
+            },
+        )?;
+
+        Ok((a_alloc.unwrap(), b_alloc.unwrap()))
+    }
+
+    // Writes `c` into the next available row's left column cell; enabling the `s_pub` selector
+    // enforces that `c = PI` for a PI provided by the verifier via the instance column `pub_col`.
+    fn alloc_public_input(&self, mut layouter: impl Layouter<Fp>, c: Fp) -> Result<Alloc, Error> {
+        let config = self.config();
+        let mut c_alloc: Option<Alloc> = None;
+
+        layouter.assign_region(
+            || "expose public input",
+            |mut region: Region<'_, Fp>| {
+                let row_offset = 0;
+                config.s_pub.enable(&mut region, row_offset)?;
+                let l_cell = region.assign_advice(
+                    || "public input advice",
+                    config.l_col,
+                    row_offset,
+                    || Ok(c),
+                )?;
+                c_alloc = Some(Alloc{ cell: l_cell, value: c});
+                Ok(())
+            },
+        )?;
+
+        Ok(c_alloc.unwrap())
+    }
+
+    // In the next available row, copies a previously allocated value `prev_alloc` into the row's left
+    // and right cells, then writes the product of the left and right cells into the row's output
+    // cell; enabling `s_mul` in the row enforces that the left, right, and output cells satisfy the
+    // multiplication constraint: `l * r = o`.
+    fn square(
+        &self,
+        mut layouter: impl Layouter<Fp>,
+        prev_alloc: Alloc,
+    ) -> Result<Alloc, Error> {
+        let config = self.config();
+        let mut squared_alloc: Option<Alloc> = None;
+        let squared_value = prev_alloc.value * prev_alloc.value;
```

Comment:
Since `Alloc` now stores values as `Option<Fp>`, calculations need to be mapped:
```suggestion
        let squared_value = prev_alloc.value.map(|prev| prev * prev);
```